pkg/service/backup_backend_s3.go

```diff
@@ -11,20 +12,22 @@ import (
 // saving state to AWS S3.
 type BackupBackendS3 struct {
 	*S3Context
-	stateFilePath string
-	backupPolicy  *model.BackupPolicy
+	stateFilePath        string
+	backupPolicy         *model.BackupPolicy
+	fullBackupInProgress *atomic.Bool
```

Comment:
Make it non-pointer and no need to pass it in the constructor.

Reply:
That's exactly why I use pointer: I need same value in BackupBackend and BackupHandler.